src/main.zig

```diff
@@ -3463,7 +3463,10 @@ fn buildOutputType(
         // incremental cache mode is used for LLVM backend too.
         if (create_module.resolved_options.use_llvm) break :b .whole;
 
-        break :b .incremental;
+        // TODO: eventually, this default should be `.incremental`. However, since incremental
+        // compilation is currently an opt-in feature, it makes a strictly worse default cache mode
+        // than `.whole`.
+        break :b .whole;
```

Comment:
```suggestion
        // Eventually, this default should be `.incremental`. However, since incremental
        // compilation is currently an opt-in feature, it makes a strictly worse default cache mode
        // than `.whole`.
        // https://github.com/ziglang/zig/issues/21165
        break :b .whole;
```